File: 프로그래머스/unrated/136798. 기사단원의 무기/기사단원의 무기.js
function solution(number, limit, power) {
    return Array.from(new Array(number), (v, i) => i + 1).reduce((acc, num) => {
        let divisorsCount = 0;
        const square = Math.sqrt(num);
        for (let i = 1; i <= Math.floor(square); i++) {
            if (num % i === 0) divisorsCount++;
        }
        const attack = Number.isInteger(square) ? divisorsCount * 2 - 1 : divisorsCount * 2;
        return attack > limit ? acc + power : acc + attack;
    }, 0)
}

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.py
def getNumOfDivisor(num):
    count = 0
    for i in range(1, int(num ** 0.5) + 1): # 제곱근까지만 반복
        if num % i == 0:
            count += 1
            if i != num // i:  # i와 num // i가 같지 않으면 짝이 있는 약수이므로
                count += 1
    return count

def solution(number, limit, power):
    iron = 0
    for i in range(1, number + 1):
        numOfDivisor = getNumOfDivisor(i)
        if numOfDivisor > limit:
            iron += power
        else:
            iron += numOfDivisor
            
    return iron